Assets/Scripts/UI/StaticScore.cs

```diff
@@ -0,0 +1,22 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+
+namespace UI
+{
+    /// <summary>
+    /// Can display the player's current score in any textbox in the application by
+    /// pulling from the score manager.
+    /// </summary>
+    public class StaticScore: MonoBehaviour
+    {
+        /// <summary>
+        /// The score text box to update and display.
+        /// </summary>
+        public Text scoreTextBox;
+
+        void Start()
+        {
+            scoreTextBox.text = "Final Score:\n" + Score.score;
```

Comment:
Reading from a static field here means that we're always going to have to have the `Score.cs` script attached to something, but not have it attached to more than one thing, since I'm not sure what the behavior would be if we had multiple `Score.cs` scripts with different scores running.
I like the direction this architecture is going, but I think we should go all the way and actually have a class called `ScoreManager` that takes care of all score-related things. We can have the in-game UI text box, as well as the end-game UI text box read from the manager. And we can have a single instance of `ScoreManager` attached to an empty game object. This is basically what I've done with `SlipTimeManager`.

Reply:
OK, that makes sense to me.

Reply:
This makes sense and I am okay with going about this direction for now. I think it is worth having `ScoreManager` and have Score script as a component attached to it so that it can update the UI. Then inside the final menu, you can just use `Score` but no need for a `ScoreManager`. `ScoreManager` will contain score-related win-lose and other updating logic.